File: src/main/java/bankaccount/history/DepositLine.java
package bankaccount.history;

import bankaccount.Amount;
import bankaccount.OperationType;

import java.time.LocalDate;

public class DepositLine extends HistoryLine {
    private DepositLine(OperationType operationType, Amount amount, LocalDate date, Amount currentBalance) {
        super(operationType, amount, date, currentBalance);
    }

    public static HistoryLine of(OperationType deposit, Amount amount, LocalDate operationDate, Amount balance) {
        return new DepositLine(deposit, amount, operationDate, balance);
    }

    public Amount combineAmounts(Amount amount) {
        return this.amount.add(amount);
    }
}

File: src/main/kotlin/bankaccount/bankaccount/DepositLine.kt
package bankaccount.bankaccount

import java.time.LocalDate

class DepositLine private constructor(amount: Amount, date: LocalDate, currentBalance: Amount) :
    HistoryLine(amount, date, currentBalance) {
    override fun printOperationType(): String {
        return "DEPOSIT"
    }

    override fun combineAmounts(amount: Amount): Amount {
        return this.amount.add(amount)
    }

    companion object {
        fun of(amount: Amount, operationDate: LocalDate, balance: Amount): HistoryLine {
            return DepositLine(amount, operationDate, balance)
        }
    }
}
